src/components/categories/Category.js

```diff
@@ -1,19 +1,64 @@
 import React from 'react';
 import { withRouter, Link } from 'react-router-dom'
 import Categories from './Categories';
+import { confirmAlert } from 'react-confirm-alert';
 import './Categories.css'
 
-class Category extends React.Component {
+  class Category extends React.Component {
+
+    getAllCategories = () => {
+      return fetch("http://localhost:8088/categories")
+      .then(res => res.json())
+      .then(res => {
+        this.setState({ categories: res  })
+      })
+  }
+
+    handleClickDelete = () => {
+      const { categories, getAllCategories   } = this.props;
+      return fetch(`http://localhost:8088/categories/${categories.id}`, {
+        method: "DELETE"
+      }).then(() => {
+        getAllCategories()
+      })
+    }
+  
+    submit = () => {
+      confirmAlert({
+        customUI: ({ onClose }) => {
+          return (
+            <div className='custom-ui'>
+              <h1>Are you sure?</h1>
+              <p>You want to delete this post?</p>
+              <button className ="mr-3 dialog-btn" onClick={onClose}><h5 className="dialog-txt">No</h5></button>
+              <button className ="dialog-btn"
+                onClick={() => {
+                  this.handleClickDelete();
+                  onClose();
+                }}
+              >
+                <h5 className="dialog-txt">Yes, Delete</h5>
+              </button>
+            </div>
+          );
+        }
+      });
+    };
+
+    
+
+
   render() {
-    const { categories } = this.props;
+    const { categories, getAllCategories } = this.props
```

Comment:
You can remove the getAllCategories variable here, because we're using up in the `handleClickDelete` function and not here in the render function.